core/libp2p/multi/multibase/codecs/base16.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <algorithm>
+#include <cctype>
+
+#include "common/hexutil.hpp"
+#include "libp2p/multi/multibase/codecs/base16.hpp"
+
+namespace {
+  /**
+   * Check, if hex string is actually in upper or lowercase
+   * @param string to be checked
+   * @param encoding_is_upper - true, if string is to be in uppercase, false
+   * otherwise
+   * @return result of the check
+   */
+  bool encodingCaseCorrect(std::string_view string, bool encoding_is_upper) {
```

Comment:
Wouldn't it be a little more neat if to remove the boolean flag and name the function like 'encodingCaseUpper'? In this case, checking if it's lower would be just a negation of the result. I mean, sometimes it's better to avoid boolean flags in function arguments, because they may be confusing in function calls.

Reply:
Absolutely, for some reason did not consider it